libtransmission/web.cc

```diff
@@ -592,6 +592,11 @@ class tr_web::Impl
             (void)curl_easy_setopt(e, CURLOPT_COOKIEFILE, file.c_str());
         }
 
+        if (auto const& proxyUrl = mediator.proxyUrl().value_or(""); !std::empty(proxyUrl))
+        {
+            (void)curl_easy_setopt(e, CURLOPT_PROXY, proxyUrl.data());
```

Comment:
```suggestion
        if (auto const& proxyUrl = mediator.proxyUrl(); proxyUrl)
        {
            (void)curl_easy_setopt(e, CURLOPT_PROXY, std::data(*proxyUrl));
```

Reply:
In my code, I try escape to call a function `curl_easy_setopt` in case when `proxy_url` is empty string (in 90% cases).

In `CURL` documentation we had next:
> Setting the proxy string to "" (an empty string) explicitly disables the use of a proxy, even if there is an environment variable set for it.

I think we should prefer next behaviour:
1. If option is not setting then pass null
2. If option is setting then pass value (empty string or not)

Reply:
Ok, makes sense to me. I can agree with keeping the current code.

Having said that, I think this needs to be explained in the `proxy_url` docs, because there's no mention about how CURL's environment variables can affect the behaviour even when `proxy_url` is set to an empty string.

Reply:
I was wrong. We should not pass `null`, because it turn off using proxy.

I think we should change curl-option only if the user set some nonempty value (option is present and value is not empty string).

Reply:
I was wrong. We should not pass `null`, because it turn off using proxy.

I think we should change curl-option only if the user set some nonempty value (option is present and value is not empty string).

Reply:
> 1. If option is not setting then pass null

Just a small clarification: You can't have "unset" settings in Transmission, the code will give it the default value.

> I think we should change curl-option only if the user set some nonempty value (option is present and value is not empty string).

Yep, so in other words, keep the current config.

Reply:
Could you explain me, please: "current code" is my or your changes? :)

If mine I shall agree keep it

Reply:
Honestly, I don't know what I should do now for PR to be continued. Could you help me, please?

Reply:
I am waiting for your comment on https://github.com/transmission/transmission/pull/5038#discussion_r1855234929.